daemons/dss-index/app.py

```diff
@@ -42,14 +42,12 @@ def dispatch_gs_indexer_event(event, context):
 
 
 def _handle_event(replica, event, context):
-    timeout = context.get_remaining_time_in_millis() / 1000 - 10  # ten seconds of safety for lambda shut down
-    assert timeout >= 10, 'Not enough time left for this lambda to process event'
     executor = ThreadPoolExecutor(len(DEFAULT_BACKENDS))
     # We can't use ecxecutor as context manager because we don't want the shutdown to block
     try:
-        backend = CompositeIndexBackend(executor, DEFAULT_BACKENDS, timeout=timeout, context=context)
+        backend = CompositeIndexBackend(executor, DEFAULT_BACKENDS)
```

Comment:
Is `executor` only used in `backend._delegate`?  I'm curious why it's passed in here.

Reply:
`_delegate` is a protected method that should not be called from outside the class. The backend methods that call it are public but their signature is restricted by the abstract class (BTW: this is an instance of the Composite design pattern). The lifetimes of the backend and the executor should be the same and providing the executor at construction time makes that explicit.